lib/puppet/provider/rabbitmq_plugin/rabbitmqplugins.rb

```diff
@@ -35,6 +40,10 @@ def destroy
   end
 
   def exists?
-    run_with_retries { rabbitmqplugins('list', '-E', '-m') }.split(%r{\n}).include? resource[:name]
+    cmd = ['list', '-E', '-m']
+    # This argument silences some extra output that's new in 3.10, but the
+    # argument is not available in some older versions
+    cmd << '-q' if Puppet::Util::Package.versioncmp(self.class.rabbitmq_version, '3.10') >= 0
```

Comment:
should this command really be running separately on both `exists?` / `self.instances?` I think there's a cleaner way based on some other providers, but not totally sure the easiest way to do that.

If not, is there another way to DRY up this logic a little?